Modules/OpenGhostDir.cs

```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
     {
 
         private static MenuButtonHolder ghostButton;
+        private static bool jobArchive;
         private static MelonPreferences_Entry<bool> _setting_GhostButton;
 
         public OpenGhostDir() =>
             _setting_GhostButton = NeonLite.Config_NeonLite.CreateEntry("Open Ghost Directory Button", true, description: "Shows a button at the end to open this level's ghost directory in the file explorer.");
 
 
-        public static string GetGhostDirectory()
+        public static string GetGhostDirectory(LevelData level = null)
         {
-            string path = GhostRecorder.GetCompressedSavePathForLevel(Singleton<Game>.Instance.GetCurrentLevel());
+            if (level == null)
+                level = Singleton<Game>.Instance.GetCurrentLevel();
+            string path = GhostRecorder.GetCompressedSavePathForLevel(level);
             path = Path.GetDirectoryName(path) + "/";
             return path;
         }
 
+        private static void SetupButton(MenuButtonHolder button, LevelData level = null)
+        {
+            try
+            {
+                if (ghostButton != null)
+                {
+                    if (jobArchive)
+                    {
+                        UnityEngine.Object.Destroy(ghostButton.gameObject);
+                        UnityEngine.Object.Destroy(ghostButton.transform.parent.gameObject);
+                        ghostButton = null;
+                    }
+                }
+
+                if (ghostButton == null)
```

Comment:
There is propably a way to initialize ghostButton once.

Reply:
As it currently stands, the `ghostButton` variable is tossed between the one in job archive and the one at stage ending. Next commit will fix this by just searching for it, as it'll be a sibling to the back button.